### PR TITLE
Explicit order in case the database returns the records in different order.

### PR DESCRIPTION
**Pull request by cincuranet**
Explicit order in case the database returns the records in different order (i.e. Firebird returns different order).

### CODE REVIEW

File: test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs [hunk 1]
```diff
@@ -2046,7 +2046,7 @@ public virtual async Task Like_with_non_string_column_using_ToString(bool async)
             var query = context.Set<Order>().Where(o => EF.Functions.Like(o.OrderID.ToString(), "%20%"));
             var result = async ? await query.ToListAsync() : query.ToList();
 
-            Assert.Equal(new[] { 10320, 10420, 10520, 10620, 10720, 10820, 10920, 11020 }, result.Select(e => e.OrderID));
+            Assert.Equal(new[] { 10320, 10420, 10520, 10620, 10720, 10820, 10920, 11020 }, result.Select(e => e.OrderID).OrderBy(x => x));
```
Comment (maumar) [inline test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs:2049]: AssertQuery infra takes care of this automatically. This test can be rewritten as:

```
            return AssertQuery(
                async,
                ss => ss.Set<Order>().Where(o => EF.Functions.Like(o.OrderID.ToString(), "%20%")),
                ss => ss.Set<Order>().Where(o => o.OrderID.ToString().Contains("20")),
                entryCount: 8);
```

File: test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs [hunk 2]
```diff
@@ -2058,7 +2058,7 @@ public virtual async Task Like_with_non_string_column_using_double_cast(bool asy
             var query = context.Set<Order>().Where(o => EF.Functions.Like((string)(object)o.OrderID, "%20%"));
             var result = async ? await query.ToListAsync() : query.ToList();
 
-            Assert.Equal(new[] { 10320, 10420, 10520, 10620, 10720, 10820, 10920, 11020 }, result.Select(e => e.OrderID));
+            Assert.Equal(new[] { 10320, 10420, 10520, 10620, 10720, 10820, 10920, 11020 }, result.Select(e => e.OrderID).OrderBy(x => x));
```
Comment (maumar) [inline test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs:2061]: same here

**Comment by cincuranet** [comment]
I was trying to keep the original test structure. Do you want me to do the change to `AssertQuery`?

**Comment by maumar** [comment]
@cincuranet yes, this will handle ordering and also allows for better re-use tests in the future. Long term goal is to convert all the query tests (that we can) to AssertQuery

File: test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs [hunk 3]
```diff
@@ -2039,26 +2039,24 @@ public virtual Task Filter_non_nullable_value_after_FirstOrDefault_on_empty_coll
...
-            using var context = CreateContext();
-
-            var query = context.Set<Order>().Where(o => EF.Functions.Like((string)(object)o.OrderID, "%20%"));
-            var result = async ? await query.ToListAsync() : query.ToList();
-
-            Assert.Equal(new[] { 10320, 10420, 10520, 10620, 10720, 10820, 10920, 11020 }, result.Select(e => e.OrderID).OrderBy(x => x));
+            return AssertQuery(
+                async,
+                ss => ss.Set<Order>().Where(o => EF.Functions.Like((string)(object)o.OrderID, "%20%")),
+                ss => ss.Set<Order>().Where(o => ((string)(object)o.OrderID.ToString()).Contains("20")),
```
Comment (maumar) [inline test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs:2058]: nit: you don't need the double cast for the expected query can just use 

```
ss => ss.Set<Order>().Where(o => o.OrderID.ToString().Contains("20"))
```

Comment (cincuranet) [reply test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs:2058]: I'm trying to keep both consistent.

Comment (maumar) [reply test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs:2058]: when we need to provide expected query (as in, when the infra can't write it for us) we tend to make it as simple/readable as possible, as long as they always produce correct expected values.

Comment (cincuranet) [reply test/EFCore.Specification.Tests/Query/NorthwindWhereQueryTestBase.cs]: OK.

**Review by maumar — APPROVED** [review]

**Comment by maumar** [comment]
@cincuranet thank you for the contribution!

**Comment by Saibamen** [comment]
Commit https://github.com/dotnet/efcore/pull/24234/commits/8cc672a4486e8e015b011789db19dea21d7a03d6 was merged or not? In `File changes` for this PR I see only 1 change (tests)

**Pull request merged**